gnomad_qc/v3/create_release/get_het_nonref_variants_to_fix.py

```diff
@@ -0,0 +1,183 @@
+) -> None:
+    """
+    Filter to variants where homalt hotfix incorrectly adjusts het nonref genotype calls.
+
+    :param hl.MatrixTable mt: Raw, split MatrixTable annotated with original genotype het nonref status.
+    :param hl.Table info_ht: Info Hail Table containing AS_lowqual information.
+    :param hl.Table freq_ht: Hail Table containing 455K frequency information.
+    :return: None
+    """
+
+    logger.info("Removing low qual variants and star alleles...")
+    mt = mt.filter_rows(
+        (~info_ht[mt.row_key].AS_lowqual)
+        & ((hl.len(mt.alleles) > 1) & (mt.alleles[1] != "*"))
+    )
+
+    logger.info("Filtering to common (AF > 0.01) variants...")
+    mt = mt.annotate_rows(AF=freq_ht[mt.row_key].freq[0].AF)
+    mt = mt.filter_rows(mt.AF > 0.01)
+
+    logger.info(
+        "Filtering to variants with at least one het nonref call and checkpointing..."
+    )
+    mt = mt.filter_rows(hl.agg.sum(mt.het_non_ref) >= 1)
+
+    logger.info(
+        "Adding new genotype entries with original homalt hotfix and het nonref-aware hotfix..."
+    )
+    mt = mt.annotate_entries(
+        GT_hotfix=hl.if_else(
+            mt.GT.is_het() & (mt.AF > 0.01) & (mt.AD[1] / mt.DP > 0.9),
+            hl.call(1, 1),
+            mt.GT,
+        ),
+        GT_hetnonref_fix=hl.if_else(
+            mt.GT.is_het()
+            & ~mt.het_non_ref
+            & (mt.AF > 0.01)
+            & (mt.AD[1] / mt.DP > 0.9),
+            hl.call(1, 1),
+            mt.GT,
+        ),
+    )
+    logger.info("Checking homozygote counts using the different genotypes...")
+    # Taking index 1 of gt_stats because it contains an entry for the reference allele at index 0
+    # homozygote_count (tarray of tint32) -
+    # Homozygote genotype counts for each allele, including the reference. Only diploid genotype calls are counted.
+    # (from hail docs, https://hail.is/docs/0.2/aggregators.html#hail.expr.aggregators.call_stats)
+
+    mt = mt.annotate_rows(gt_stats=hl.agg.call_stats(mt.GT_hotfix, mt.alleles))
+    mt = mt.transmute_rows(hom_count_hotfix=mt.gt_stats.homozygote_count[1])
+
+    mt = mt.annotate_rows(gt_stats=hl.agg.call_stats(mt.GT_hetnonref_fix, mt.alleles))
+    mt = mt.transmute_rows(hom_count_hetnonref_fix=mt.gt_stats.homozygote_count[1])
+
+    logger.info(
+        "Filtering to rows where homalt hotfix erroneously adjusts het nonref calls..."
+    )
+    mt = mt.filter_rows(mt.hom_count_hotfix != mt.hom_count_hetnonref_fix)
+
+    return mt.rows()
+
+
+def main(args):
+    """
+    Script used to get variants impacted by homalt hotfix.
+    Most of the work was done in notebooks, and any results created in notebooks are noted with comments.
+    """
+
+    hl.init(log="/get_impacted_variants.log", default_reference="GRCh38")
+
+    logger.info("Reading sparse MT and metadata table with release only samples...")
+    mt = get_gnomad_v3_mt(
+        key_by_locus_and_alleles=True, samples_meta=True, release_only=True,
```

Comment:
Yes, but we are handling that subset in a different script now so this will actually only patch the full release since we actually need to redo all frequencies for the HGDP + TGP subset

Reply:
code for this subset has been added and frequency recalculated with the fix already put in, so marking resolved